File: projects/integration/transforms-python/src/python/exposure/current/fp_gas_m0145.py
# ===== import: python function

# ===== import: palantir functions
from transforms.api import transform_df, Input, Output, configure

# ===== import: our functions
from python.util.bp_dateutils import filter_latest_runs
from python.util.bp_config import Config
from python.util.bp_constants import LOOK_BACK_PERIOD_CONFIG
from python.bp_flush_control import FPGAS145_FLUSH, FPGAS145_FLUSH_START_DATE, FPGAS145_PROFILE


def convert(df):
    config = Config(look_back_period=LOOK_BACK_PERIOD_CONFIG,
                    flush=FPGAS145_FLUSH,
                    flush_start_date=FPGAS145_FLUSH_START_DATE)
    df = df.filter((df.Analysis_Date >= config.start_date) & (df.Analysis_Date <= config.end_date))
    df = filter_latest_runs(df, 'Analysis_Date', 'Run_Datetime')
    return df


@configure(profile=FPGAS145_PROFILE)
@transform_df(
    Output("/BP/IST-IG-DD/data/technical/exposure/current/freeport/fp_gas_m0145"),
    df=Input("/BP/IST-IG-DD/data/technical/exposure/typed/freeport/fp_gas_m0145"),
)
def my_compute_function(df):
    """
    Transformation stage function
    Version: T2
    This function:
        For processing current data.
        Reorders the dataframe based on its respective output schema

    Args:
        df (dataframe): current fp_gas_145m

    Returns:
        dataframe: transformed fp_gas_145m
    """
    return convert(df)

File: projects/IST-IG-DD-JAVA/transforms-java/src/main/java/myproject/datasets/exposure/current/fp_gas_m0145.java
package myproject.datasets.exposure.current;

import com.palantir.transforms.lang.java.api.Compute;
import com.palantir.transforms.lang.java.api.Input;
import com.palantir.transforms.lang.java.api.Output;
import myproject.datasets.util.BP_DateUtils;
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;

import java.util.Date;

public class fp_gas_m0145 {
    public Dataset<Row> convert(Dataset<Row> df) {
        Date[] dates = BP_DateUtils.getStartEndDate();
        df = BP_DateUtils.filterLatest(df, dates[0], dates[1], "Analysis_Date", "Run_Datetime");
        return df;
    }

    @Compute
    @Output("/BP/IST-IG-DD/data/technical/exposure_java/current/freeport/fp_gas_m0145")
    public Dataset<Row> myComputeFunction(
            @Input("/BP/IST-IG-DD/data/technical/exposure_java/typed/freeport/fp_gas_m0145") Dataset<Row> df
    ){
        return convert(df);
    }

}
